### ISSUE TITLE
Convert parser warnings to lints?

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2065]
We already convert parser errors to lints (`type = "error"`), but what about parser warnings?

I came across such a warning when trying to lint `r-devel`, it came from a test file where the lint was intentional:

```r
lintr::lint("tests/simple-true.R", linters = list())
# Warning messages:
# 1: In parse(text = source_expression$content, srcfile = source_expression,  :
#   non-integer value 1e-3L qualified with L; using numeric value
# 2: In parse(text = source_expression$content, srcfile = source_expression,  :
#   integer literal 1.L contains unnecessary decimal point
# 3: In parse(text = source_expression$content, srcfile = source_expression,  :
#   non-integer value 1e-3L qualified with L; using numeric value
# 4: In parse(text = source_expression$content, srcfile = source_expression,  :
#   integer literal 1.L contains unnecessary decimal point
```

Should those be returned as lints (`type = "warning"`) instead?

**Comment by MichaelChirico** [comment]
Searching gram.y, it seems all of the possible warnings from parsing are from issues with numeric literals:

https://github.com/r-devel/r-svn/blob/c9934dd660808be35bd225a89af4b6f4b4086a29/src/main/gram.y#L2655-L2670

This should help with generatic lint metadata.

**Comment by MichaelChirico** [comment]
One issue is that the generated warnings don't contain any source location metadata:

```r
parse(text = R"(
1e-3L
1e-3L
1.0L
1.0L
1.0e-3L
1.0e-3L
)")
```

```
Warning messages:
  non-integer value 1e-3L qualified with L; using numeric value
  non-integer value 1e-3L qualified with L; using numeric value
  integer literal 1.0L contains unnecessary decimal point
  integer literal 1.0L contains unnecessary decimal point
  non-integer value 1.0e-3L qualified with L; using numeric value
  non-integer value 1.0e-3L qualified with L; using numeric value
```

I guess we can just `grep()` through the content.

**Comment by MichaelChirico** [comment]
I filed https://bugs.r-project.org/show_bug.cgi?id=18863 to eventually have R itself provide the location info.
